bot/exts/evergreen/fun.py

```diff
@@ -87,6 +92,90 @@ def conversion_func(text: str) -> str:
             converted_text = f">>> {converted_text.lstrip('> ')}"
         await ctx.send(content=converted_text, embed=embed)
 
+    @commands.group(name="caesarcipher", aliases=("caesar",))
+    async def caesarcipher_group(self, ctx: Context) -> None:
+        """
+        Translates a message using the Caesar Cipher.
+
+        See `decrpyt`, `encrypt`, and `info` subcommands.
+        """
+        if ctx.invoked_subcommand is None:
+            await ctx.invoke(self.bot.get_command("help"), "caesarcipher")
+
+    @caesarcipher_group.command(name="info")
+    async def caesarcipher_info(self, ctx: Context) -> None:
+        """Information about the Caesar Cipher."""
+        embed = Embed.from_dict(self._caesar_cipher_embed)
+        embed.colour = Colours.dark_green
+
+        await ctx.send(embed=embed)
+
+    @staticmethod
+    async def _caesar_cipher(ctx: Context, offset: int, msg: Union[Message, str]) -> None:
+        """
+        Given an integer `offset`, translates and sends the given `text`.
+
+        A positive `offset` will cause the letters to shift right, while
+        a negative `offset` will cause the letters to shift left.
+
+        Also accepts a valid Discord Message ID or link.
+        """
+        def caesar_func(text: str) -> str:
```

Comment:
I've recently noticed this has an incorrect annotation for its return type. Should this be corrected to `Generator[str, None, None]`?

Reply:
I think `Iterator[str]` ought be enough instead of the whole Generator typehint when it only yields

Reply:
I've seen people annotate generators using just the yield type (such as here), but yeah using `Iterator[str]` would probably be best. The [docs](https://docs.python.org/3/library/typing.html#typing.Generator) mention it as a valid alternative to `Generator[str, None, None]`.